frontend/src/pages/groups/GroupCreatePage.tsx

```diff
@@ -10,27 +11,35 @@ import { groupViewRoute } from '../../utils/routes'
 import GroupForm from './GroupForm'
 
 const GroupCreatePage: FunctionComponent = () => {
+  const { refetch } = useContext(UserProfileContext)
   const history = useHistory()
 
   const [
     addGroup,
     { loading: mutationIsLoading, error: mutationError },
   ] = useCreateGroupMutation()
 
-  const onSubmit = (input: GroupCreateInput) => {
-    // Create the group and then redirect to its view/edit page
-    addGroup({
-      variables: { input },
-      // Fetch the updated list of groups
-      refetchQueries: [{ query: AllGroupsDocument }],
-    })
-      .then(({ data }) => {
-        if (data) {
-          const newGroupId = data.addGroup.id
-          history.push(groupViewRoute(newGroupId))
-        }
+  const onSubmit = async (input: GroupCreateInput) => {
+    try {
+      // Create the group and then redirect to its view/edit page
+      const { data } = await addGroup({
+        variables: { input },
+        // Fetch the updated list of groups
+        refetchQueries: [{ query: AllGroupsDocument }],
       })
-      .catch(console.error)
+
+      // If the user is a group leader, refresh their credentials so that the UI
+      // is aware that they created their own group and finished their
+      // onboarding
```

Comment:
Does this make sense/is it clear enough?

Reply:
I'm not clear on why/how `refetch` does this. The comment beginning with "If" leads me to expect a conditional.

Suggestion: add a sentence that begins with "refetch here will..."